### ISSUE TITLE
async job test_delete test failing

### ISSUE THREAD

**Issue by murphyke** [chop-dbhi/serrano#299]
TL;DR - I think this is a minor Avocado issue.

```
======================================================================
FAIL: test_delete (tests.cases.resources.tests.async.base.JobTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/murphyke/Documents/code/python/serrano-env/serrano-murphyke/tests/cases/resources/tests/async/base.py", line 82, in test_delete
    self.assertEqual(response.status_code, codes.not_found)
AssertionError: 200 != 404
```

The statement [self.client.get('/api/jobs/{0}/')](https://github.com/chop-dbhi/serrano/blob/master/tests/cases/resources/tests/async/base.py#L80-L81) is expected to return 404 but returns 200.

Earlier in the test, the statement [self.client.delete('/api/jobs/{0}/')](https://github.com/chop-dbhi/serrano/blob/master/tests/cases/resources/tests/async/base.py#L74-L75) removes the job from the queue but *does not remove it from Redis*.  Ultimately this is due to Avocado's use of [rq.Job.cancel](python test_suite.py resources.JobTestCase.test_delete) instead of [rq.Job.delete](https://github.com/nvie/rq/blob/6a57d321f2489266ccfe9ddb7b0ff5f14fce1e2f/rq/job.py#L483-L489).

Then [self.client.get('/api/jobs/{0}/')](https://github.com/chop-dbhi/serrano/blob/master/tests/cases/resources/tests/async/base.py#L80-L81) is supposed to not find the job and fail with a 404. However, the queue is never consulted.  The (undocumented) [rq.Queue.fetch_job](https://github.com/nvie/rq/blob/6a57d321f2489266ccfe9ddb7b0ff5f14fce1e2f/rq/queue.py#L108-L112) is used, which, amazingly, never actually looks at the queue; it just plucks the job from Redis.  This was [fixed](https://github.com/nvie/rq/issues/728) 8 days ago, but the fix hasn't made it into a release.  When I tested against RQ master, the Serrano test still failed.

If [avocado.async.utils](https://github.com/chop-dbhi/avocado/blob/master/avocado/async/utils.py#L84) is changed to delete the job instead of canceling, the Serrano test passes.  I haven't looked at it, but the [avocado cancel_all_jobs function](https://github.com/chop-dbhi/avocado/blob/master/avocado/async/utils.py#L88-L92) may also empty the queue without purging jobs from Redis.

I'll submit a PR against Avocado.

On the Serrano side, I don't think any changes are needed.

**Comment by bruth** [comment]
Brilliant. Great investigative work.
